Feat: Add notification dropdown

hannadoolsetnet opened the pull request:
## 알림 리스트 dropdown

알림 버튼 클릭 시 로그인 상태인 사용자의 알림 리스트 중 읽지 않은 알림의 목록을 보여주는 dropdown입니다.

### <특이사항>

- 현재 알림 리스트의 각 목록을 클릭해야 dropdown이 종료되는 상태인데, 추후 dropdown 바깥 부분을 클릭했을 때에도 종료할 수 있도록 수정이 필요합니다.

hannadoolsetnet committed:
Feat: Add notification dropdown

SeaBlueMaker commented on src/components/NotificationButton.js:
```diff
@@ -1,22 +1,45 @@
+import Dropdown from "../components/common/Dropdown";
 
 import METHODS from "../constants/methods";
+import { useState } from "react";
```
다른 컴포넌트들처럼 리액트 관련 import는 최상단에서 이루어지면 좋겠습니다!

hannadoolsetnet committed:
Refactor: Move import statement

imjaehoo commented on src/components/NotificationButton.js:
```diff
@@ -1,22 +1,45 @@
+  const [notifications, setNotifications] = useState([]);
+
   const handleButtonClick = async () => {
     const { GET } = METHODS;
```
import 할 때 처음부터 destructuring으로 GET만 가져오고 여기선 갖다 쓰기만 하면 어떨까요?
버튼을 클릭할 때마다 변수를 만들어야 할지?

hannadoolsetnet replied:
알겠습니다~!🙏🏻

imjaehoo commented on src/components/NotificationButton.js:
```diff
@@ -1,22 +1,45 @@
     const { GET } = METHODS;
 
-    await fetch(`${process.env.REACT_APP_SERVER_URL}/users/notification`, {
+    const notifications = await fetch(`${process.env.REACT_APP_SERVER_URL}/users/notification`, {
```
서버에서 값을 제대로 못 보내주면 40번째 줄에서 에러가 날 것 같네용
그리고 api 폴더가 있길래 이런 로직들을 분리하려는 목적인줄 알았는데 아닌가요?

hannadoolsetnet replied:
네네 맞습니다! 해당 부분은 리팩토링 시 분리할 예정이고, 다음부터는 바로 분리해서 작성하도록 하겠습니다!

imjaehoo commented on src/components/NotificationButton.js:
```diff
@@ -1,22 +1,45 @@
+        <img src="images/notification_icon.png" alt="알림 버튼" width="40px" height="35px" />
+      </Button>
+      {isClicked && (
+        <Dropdown name="list" onClick={handleDropdownClick}>
```
얘도 그렇고 Button도 그렇고 name이 조금 어색.. type 개념일텐데 type을 못 쓴다면 variant는 어떨까 해요.

hannadoolsetnet replied:
안 그래도 이 부분에 대해 고민이 많았는데 유익한 조언 감사합니다!🙏🏻

hannadoolsetnet committed:
Refactor: Refactor notification button component

  - move import statement
  - use destructuring
  - modify notification list button click handler
